### PR TITLE
Fixes #22205 - restore blueprint actions for notifications

### PR DESCRIPTION
**Pull request by ares**
self.actions defaults to `{}`, the logic with conditional overiding is correct but the condition is wrong

### CODE REVIEW

**Comment by tbrisker** [comment]
@ares please rebase on latest develop to fix failing minitest issue

**Comment by ares** [comment]
thanks @tbrisker, rebased

**Comment by mmoll** [comment]
another rebase for the audited pin is needed now...

**Comment by ekohl** [comment]
Does this also need the 1.17 milestone?

**Comment by ares** [comment]
it's set in redmine, but I assigned the milestone too, I'll rebase soon

**Comment by ares** [comment]
rebased

File: app/models/notification.rb [hunk 1]
```diff
@@ -66,10 +66,10 @@ def set_notification_recipients
 
   def set_custom_attributes
     return unless notification_blueprint # let validation catch this.
-    self.actions ||= UINotifications::URLResolver.new(
+    self.actions = UINotifications::URLResolver.new(
       subject,
       notification_blueprint.actions
-    ).actions if notification_blueprint.actions.any?
+    ).actions if notification_blueprint.actions.any? && self.actions.blank?
```
Comment (ekohl) [inline app/models/notification.rb:72]: Not a ruby developer, but multi line statements with guards are hard to parse for me.

**Review by boaz0 — APPROVED** [review]
LGTM 💯

File: app/models/notification.rb (hunk 1 above)
Comment (ares) [reply app/models/notification.rb:72]: I'm not making the problem worse, since this will be cherrypicked I'd probably keep it as it is, I can open separate PR with refactoring if you prefer

Comment (ekohl) [reply app/models/notification.rb:72]: Sounds good.

Comment (ares) [reply app/models/notification.rb:72]: https://github.com/theforeman/foreman/pull/5170 addresses it

**Review by tbrisker — APPROVED** [review]
Thanks @ares !

**Comment by tbrisker** [comment]
Cherry-picked to 1.17-stable as well as d11899e8425af20e14cc9bbed363c7d63e5892df
